src/benchmarklib/benchmark_runner.cpp

```diff
@@ -264,36 +274,33 @@ void BenchmarkRunner::_benchmark_shuffled() {
     item_ids = item_ids_weighted;
   }
 
-  auto item_ids_shuffled = std::vector<BenchmarkItemID>{};
-
   for (const auto& item_id : item_ids) {
     _warmup(item_id);
   }
 
   // For shuffling the item order.
-  std::random_device random_device;
-  std::mt19937 random_generator(random_device());
+  auto random_device = std::random_device{};
+  auto random_generator = std::mt19937{random_device()};
+
+  std::shuffle(item_ids.begin(), item_ids.end(), random_generator);
+  auto item_id_counter = size_t{0};
 
   Assert(_currently_running_clients == 0, "Did not expect any clients to run at this time.");
 
   _state = BenchmarkState{_config.max_duration};
+  _running_clients_semaphore.release(_config.clients);
 
   while (_state.keep_running() && (_config.max_runs < 0 || _total_finished_runs.load(std::memory_order_relaxed) <
```

Comment:
Not yours, just adding here to be able a comment: I am not 100% sure if `keep_running` vs. `is_done` in `_schedule_item_run()` is always correct, esp. for very short-running queries, multiple clients, and shuffled runs. Same for clients that finished after max runs finished. Remind me to discuss this offline.

Reply:
As discussed, won't be addressed in this PR.